runtime/lua/vim/img.lua

```diff
@@ -0,0 +1,78 @@
+local img = vim._defer_require('vim.img', {
+  _backend = ..., --- @module 'vim.img._backend'
+  _detect = ..., --- @module 'vim.img._detect'
+  _image = ..., --- @module 'vim.img._image'
+  _terminal = ..., --- @module 'vim.img._terminal'
```

Comment:
Is there a need for this to be defined on `vim`? usually code should just `require()` stuff, just-in-time if it wants "lazy" behavior.

This `vim._defer_require` is for things that we actually need on `vim.*`.

Reply:
Cool, I wasn't sure the best practice here. I just looked at how `vim.lsp` was done and copied it. So I can switch to regular `require` instead.